util/yang.go

```diff
@@ -361,6 +362,37 @@ func EnumeratedUnionTypes(types []*yang.YangType) []*yang.YangType {
 	return eTypes
 }
 
+// EnumeratedUnionSubsumingTypes returns the subsuming types of the enumerated
```

Comment:
Do you mean the subsumed types, rather than the subsuming types? i.e., we're walking from a parent downwards so everything that is below us is presumably subsumed by the supplied parent -- whereas subsuming types rather sounds like we're walking up from a child to find all of the parent types that include us.

Reply:
I do mean the subsuming types. It's just that the child types don't have a parent pointer, so the implementation currently has to go downwards.

Reply:
I renamed and simplified the semantics of this function. Hopefully it's not confusing anymore.